facilitator/src/transport.rs

```diff
@@ -15,9 +14,28 @@ use std::{
 use uuid::Uuid;
 
 pub use self::s3::S3Transport;
+use self::{gcs::GcsError, s3::S3Error};
 pub use gcs::GcsTransport;
 pub use local::LocalFileTransport;
 
+/// Common error type for I/O over any batch transport.
+#[derive(Debug, thiserror::Error)]
+pub enum TransportError {
+    #[error("object {0} not found: {1}")]
+    ObjectNotFoundError(String, anyhow::Error),
+    #[error(transparent)]
+    Local(#[from] local::FileError),
+    #[error(transparent)]
+    Gcs(#[from] GcsError),
+    #[error(transparent)]
+    S3(#[from] S3Error),
+    #[error("an error occurred while cancelling after another error: {original}, {cancellation}")]
+    Cancellation {
+        original: Box<TransportError>,
+        cancellation: Box<TransportError>,
```

Comment:
I was getting a Clippy warning about a large difference between sizes of variants elsewhere, (i.e., `BatchReadError::Transport`) but here in particular I need the boxes because a `TransportError` can't contain another `TransportError` in-line.